ambry-messageformat/src/test/java/com.github.ambry.messageformat/BlobPropertiesTest.java

```diff
@@ -44,51 +48,79 @@
    */
   @Parameterized.Parameters
   public static List<Object[]> data() {
-    return Arrays.asList(new Object[][]{{BlobPropertiesSerDe.VERSION_1}, {BlobPropertiesSerDe.VERSION_2}});
+    return Arrays.asList(
+        new Object[][]{{BlobPropertiesSerDe.VERSION_1}, {BlobPropertiesSerDe.VERSION_2}, {BlobPropertiesSerDe.VERSION_3}});
   }
 
   public BlobPropertiesTest(short version) {
     this.version = version;
   }
 
   @Test
-  public void basicTest() {
-    final int blobSize = 100;
-    final String serviceId = "ServiceId";
-    final String ownerId = "OwnerId";
-    final String contentType = "ContentType";
-    final int timeToLiveInSeconds = 144;
-
-    short accountId = UNKNOWN_ACCOUNT_ID;
-    short containerId = UNKNOWN_CONTAINER_ID;
-    if (version == BlobPropertiesSerDe.VERSION_2) {
-      accountId = Utils.getRandomShort(TestUtils.RANDOM);
-      containerId = Utils.getRandomShort(TestUtils.RANDOM);
-    }
-
-    BlobProperties blobProperties = getBlobProperties(blobSize, serviceId, accountId, containerId, version);
+  public void basicTest() throws IOException {
+    int blobSize = 100;
+    String serviceId = "ServiceId";
+    String ownerId = "OwnerId";
+    String contentType = "ContentType";
+    int timeToLiveInSeconds = 144;
+    short accountId = Utils.getRandomShort(TestUtils.RANDOM);
+    short containerId = Utils.getRandomShort(TestUtils.RANDOM);
+    boolean isEncrypted = true;
+
+    BlobPropertiesSerDe.CURRENT_VERSION = version;
+
+    short accountIdToExpect = version == BlobPropertiesSerDe.VERSION_1 ? UNKNOWN_ACCOUNT_ID : accountId;
+    short containerIdToExpect = version == BlobPropertiesSerDe.VERSION_1 ? UNKNOWN_CONTAINER_ID : containerId;
+    boolean encryptFlagToExpect = version == BlobPropertiesSerDe.VERSION_3 && isEncrypted;
+
+    BlobProperties blobProperties =
+        getBlobProperties(blobSize, isEncrypted, serviceId, accountId, containerId, version);
     System.out.println(blobProperties.toString()); // Provide example of BlobProperties.toString()
-    verifyBlobProperties(blobProperties, blobSize, serviceId, null, null, false, Utils.Infinite_Time, accountId,
-        containerId);
+    ByteBuffer serializedBuffer = ByteBuffer.allocate(BlobPropertiesSerDe.getBlobPropertiesSerDeSize(blobProperties));
```

Comment:
So, looks like we didn't have a separate test class for BlobPropertiesSerDe prior to this patch. Were we relying on MessageFormatRecordTest for coverage then ?